src/js/index.js

```diff
@@ -107,388 +110,574 @@ map.on('load', e=>{
-        .then(response => {
-            if (response.status == 200) { return response.json() }
-        })
-        .then(jawn => {
-            // create filter array for hex tile
-            let hex_values = []
-            let popupReference = new Object();
-            jawn.cargo.map(hex=>{
-                hex_values.push(hex.hex_id)
-                popupReference[hex.hex_id] = hex.count
-                stationInfo.data.push({
-                    id: hex.hex_id,
-                    count: hex.count
-                })
-            })
-            // style
-            if (map.getSource('hexBins')) {
-                let style = HexStyling(stationInfo, schemes, hex_values)
-                map.addLayer(style, 'railHighlight')
-                map.on('click', e=>{
-                    if (map.getLayer('hexClick')){
-                        map.removeLayer('hexClick')
-                    }
-                })
-
-                map.on('click', 'hexBins', e=>{
-                    if (popupReference[e.features[0].properties.FID]){
-                        let count = popupReference[e.features[0].properties.FID]
-                        let offsets = {
-                            'top': [0, 0],
-                            'top-left': [0,0],
-                            'top-right': [0,0],
-                            'bottom': [0, -15],
-                            'bottom-left': [0,0],
-                            'bottom-right': [0,0],
-                            'left': [0,0],
-                            'right': [0,0]
-                            }
-                        
-                        const popup = new mapboxgl.Popup({ offset: offsets, className: 'map__hexPopup' })
-                        let colors = schemes[stationInfo.operator][stationInfo.mode]
-                        popup.setLngLat(e.lngLat)
-                            .setHTML(`<p style="color: ${colors[colors.length-1]}">${count} Commuters attributed to this area.</p>`)
-                            .addTo(map)
-                        document.querySelector('.mapboxgl-popup-close-button').addEventListener('click', e=>{
-                            if (map.getLayer('hexClick')){
-                                map.removeLayer('hexClick')
-                            }
-                        })
-                        for (let node of document.querySelectorAll('.mapboxgl-popup-tip')){
-                            node.style.borderTopColor = colors[0]
-                        }
-                        for (let node of document.querySelectorAll('.mapboxgl-popup-content')){
-                            node.style.borderColor = colors[1]
-                        }
+map.on('click', 'railStations-base', e=>{
+    let props = e.features[0].properties
+    if (props.SURVEY_ID > 0){
+        // update station dropdown
+        let form = document.querySelector('form')
```

Comment:
now that PerformQuery is refactored into a more generic function that takes station ID and year and return the jawns, does this click handler still need to populate the form? Couldn't the function be simplified even further to something like:

map.on('click', 'railStations-base', e => {
    let props = e.features[0].properties
    if(props.SURVEY_ID > 0){
        year = data[props.SURVEY_ID].years[0]
        PerformQuery(props.SURVEY_ID, year)
    }
}

Reply:
Now that I've made that change, I'm wondering if the form should still be populated. After the query is completed via click, I think it would be good to have another reference to what is being displayed alongside the information in the legend. Thoughts? It's redundant but that's not necessarily a bad thing.